### PR TITLE
Use Windows Sleep for framerate control

### PR DESCRIPTION
**Pull request by quyykk**
## Fix Details

Due to the recent update to MinGW 11, it turns out that the Winpthread implementation that MinGW uses for the C++11 threading library sleeps too long. This seems to be a "problem" since MinGW 9 with the only solution being to use the native Windows sleep function instead of the Winpthread implementation.

Sleeping too long is technically valid behavior, but this meant that the game wasn't running at 60fps. So this PR #ifdef the sleeping call to use the Windows Sleep function.

## Testing Done

Used a frame viewer to verify that this PR fixes the framerate drops that occur using the MinGW 11 executable from Github Actions.

### CODE REVIEW

File: source/FrameTimer.cpp [hunk 1]
```diff
@@ -14,6 +14,10 @@ PARTICULAR PURPOSE.  See the GNU General Public License for more details.
 
 #include <thread>
 
+#ifdef _WIN32
+#include <Windows.h>
```
Comment (MCOfficer) [inline source/FrameTimer.cpp:18]:
```suggestion
#include <windows.h>
```
Not sure if there's a styleguide behind it, but that's how it's done in [other files](https://github.com/endless-sky/endless-sky/blob/master/source/Files.cpp#L23).

**Comment by Hecter94** [comment]
Seems to have resolved the issue, game runs at a solid 60fps while using this PR, compared to the 40fps that the current continuous build runs at. 
No other issues seen.

**Comment by tehhowch** [comment]
Part of me would rather we restored builds with MinGW 8 😅

I'm not able to verify this change is ok when using MinGW 5/8 until the 19th or later. If you verify we don't have any issues with frame rate with those (8 at least), I'm fine with this merging.

**Comment by quyykk** [comment]
> Part of me would rather we restored builds with MinGW 8 😅

I mean, at some point we will have to update. Might as well do it now.

> I'm not able to verify this change is ok when using MinGW 5/8 until the 19th or later.

I don't see what problems there could be. The only thing this PR does is call the Windows Sleep function, which is not part of mingw but part of Windows. Doesn't matter what compiler you use.

Nevertheless, I'll try it out with MinGW 8.

**Comment by tehhowch** [comment]
> > Part of me would rather we restored builds with MinGW 8 😅
> 
> I mean, at some point we will have to update. Might as well do it now.

Sure, but this regression isn't guaranteed to be present in every version to which we could upgrade in the future.

**Pull request merged**